GitUI/UserControls/RevisionGrid.cs

```diff
@@ -1687,18 +1693,18 @@ private void RevisionsCellPainting(object sender, DataGridViewCellPaintingEventA
                         int gravatarTop = e.CellBounds.Top + textHeight + 6;
                         int gravatarLeft = e.CellBounds.Left + baseOffset + 2;
 
-
-                        Image gravatar = Gravatar.GravatarService.GetImageFromCache(revision.AuthorEmail + gravatarSize.ToString() + ".png", revision.AuthorEmail, AppSettings.AuthorImageCacheDays, gravatarSize, AppSettings.GravatarCachePath, FallBackService.MonsterId);
-
-                        if (gravatar == null && !string.IsNullOrEmpty(revision.AuthorEmail))
+                        var gravatar = _avatarCache.GetImage($"{revision.AuthorEmail}.png", null);
+                        if (gravatar == null)
                         {
-                            ThreadPool.QueueUserWorkItem(o =>
-                            Gravatar.GravatarService.LoadCachedImage(revision.AuthorEmail + gravatarSize.ToString() + ".png", revision.AuthorEmail, null, AppSettings.AuthorImageCacheDays, gravatarSize, AppSettings.GravatarCachePath, RefreshGravatar, FallBackService.MonsterId));
+                            gravatar = Resources.User;
+                            // kick off download operation, will likely display the avatar during the next round of repaint
+                            await _gravatarService.GetAvatarAsync(revision.AuthorEmail, AppSettings.AuthorImageSize, AppSettings.GravatarDefaultImageType)
+                                .ContinueWith(t => this.InvokeAsync(() =>
+                                {
+                                    Revisions.InvalidateCell(e.ColumnIndex, e.RowIndex);
```

Comment:
You have to ensure that the cell is still valid. I get an error here:
```
   at System.Windows.Forms.DataGridView.InvalidateCell(Int32 columnIndex, Int32 rowIndex) 
   at GitUI.RevisionGrid.<>c__DisplayClass227_1.<RevisionsCellPainting>b__4() in C:\d\projekty\gitextensions\GitUI\UserControls\RevisionGrid.cs:line 1704
   at GitUI.GitUIExtensions.<>c__DisplayClass17_0.<InvokeAsync>b__0(Object _) in C:\d\projekty\gitextensions\GitUI\GitUIExtensions.cs:line 303
   at GitUI.GitUIExtensions.<>c__DisplayClass18_0.<InvokeAsync>b__0(Object s) in C:\d\projekty\gitextensions\GitUI\GitUIExtensions.cs:line 311
```

Is not it enough that RevisionsGrid gets invalidated on the cache changed event?